### PR TITLE
docker-push/pull fixes and adding namespace concept

### PR DESCRIPTION
**Pull request by sairez**

### CODE REVIEW

File: bin/docker-push [hunk 1]
```diff
@@ -17,7 +17,7 @@ then
   exit 1
 fi
 
-docker tag $DOCKERTAG:latest ${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:$CI_SHA1
+docker tag -f $DOCKERTAG:latest ${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:$CI_SHA1
```
Comment (rosskukulinski) [inline bin/docker-push]: So after some research, I don't think we should be using `tag -f`.  This was deprecated in Docker 1.10 and removed in 1.12.

https://docs.docker.com/engine/deprecated/

> To make tagging consistent across the various docker commands, the -f flag on the docker tag command is deprecated. It is not longer necessary to specify -f to move a tag from one image to another. Nor will docker generate an error if the -f flag is missing and the specified tag is already in use.

Could you remove this from your PR?

File: bin/k8s-apply [hunk 2]
```diff
@@ -3,35 +3,39 @@
 . k8s-read-config
 
 echo "Updating ConfigMaps"
-for i in "${CONFIGMAP_FILES[@]}"
+for index in "${!CONFIGMAP_FILES[@]}"
```
Comment (rosskukulinski) [inline bin/k8s-apply:6]: Curious as to why the loop structure is changing throughout this file.

File: bin/k8s-read-config [hunk 3]
```diff
@@ -34,6 +34,9 @@ fi
 echo "[dockerfile]: $DOCKERFILE"
 echo "[dockertag]: $DOCKERTAG"
 
+NAMESPACE=${NAMESPACE:=default}
```
Comment (rosskukulinski) [inline bin/k8s-read-config:37]: So this makes it a breaking change -- I'm not opposed, just mostly clarifying.

**Review by rosskukulinski — CHANGES_REQUESTED** [review]
Hi Sarah - thanks for adding namespace support!  My two over-arching comments:
- I don't think we should be using tag -f.  Could you please remove that?
- I'm wondering why you changes the loops to give you the index in k8s-apply, k8s-delete, k8s-deploy

File: bin/k8s-apply (hunk 2 above)
Comment (sairez) [reply bin/k8s-apply:6]: This was mainly an attempt at making looping structure consistent across scripts. I saw this structure through k8s-deploy on `master` and wondered why the two styles were used to seemingly similar effect. Not attached to the modification if you'd like to revert it.

File: bin/k8s-read-config (hunk 3 above)
Comment (sairez) [reply bin/k8s-read-config:37]: I'm guessing you mean breaking if one does not already specify `NAMESPACE` in their config? I thought setting the default here to its current assumed value would take care of that case?

**Review by rosskukulinski — APPROVED** [review]

**Pull request merged**